cpp/src/arrow/compute/util.h

```diff
@@ -89,16 +89,22 @@ class ARROW_EXPORT TempVectorStack {
   Status Init(MemoryPool* pool, int64_t size) {
     num_vectors_ = 0;
     top_ = 0;
-    buffer_size_ = PaddedAllocationSize(size) + kPadding + 2 * sizeof(uint64_t);
+    buffer_size_ = EstimateAllocSize(size);
     ARROW_ASSIGN_OR_RAISE(auto buffer, AllocateResizableBuffer(size, pool));
     // Ensure later operations don't accidentally read uninitialized memory.
     std::memset(buffer->mutable_data(), 0xFF, size);
     buffer_ = std::move(buffer);
     return Status::OK();
   }
 
+  static int64_t EstimateAllocSize(int64_t size) {
+    return PaddedAllocationSize(size) + 2 * sizeof(uint64_t);
+  }
+
+  Status CheckAllocOverflow(int64_t alloc_size);
+
  private:
-  int64_t PaddedAllocationSize(int64_t num_bytes) {
+  static int64_t PaddedAllocationSize(int64_t num_bytes) {
```

Comment:
Maybe we can align all the function names to either `XxxAllocSize` or `XxxAllocationSize`?